### PR TITLE
✨Integrate React Flow with Remix

### PR DESCRIPTION
**Pull request by shurrini**
# ✨ Integrate React Flow with Remix

## Description

React flow library was installed into remix succesfully. Additionally, libraries like shadcn and react-icons were also installed. Based on these libraries, a simple static flow was created using client-side rendering since react flow doesn't support server-side rendering.

## Related JIRA Tickets

- https://magnimussftware.atlassian.net/browse/PLANES-56

## Changes
- Added a route called "workflow" for the React Flow components.
- Installed React Flow, shadcn, and react-icons libraries, modifying package.lock.json, package.json, yarn.lock, and other files in the process.
- Implemented button and dialog components from shadcn.
- Created a static react flow with a custom edge in the workflow route.

## Checklist

- [x] I have tested my changes thoroughly.
- [x] I have updated the documentation to reflect the changes.
- [x] I have added unit tests (if applicable) to cover the changes.
- [x] My code follows the project's coding standards and style guidelines.
- [x] I have rebased my branch on the latest main/development branch.
- [x] All existing tests are passing.
- [x] I have run the linter and addressed any warnings or errors.
- [x] I have self-reviewed my code to catch potential issues.
- [x] I have considered the potential impact of these changes on other parts of the monorepo.
- [x] I have tested these changes in a local environment.

### CODE REVIEW

File: services/dashboard/workflow/buttonedge.tsx
Comment (enZane) [inline services/dashboard/workflow/buttonedge.tsx:1]: Can you move this folder to `app/components`? To have centralized all the components

File: services/dashboard/app/root.tsx [hunk 1]
```diff
@@ -201,8 +204,40 @@ function App() {
 				</header>
 
 				<div className="flex-1">
-					<p>ReactFlow Testing</p>
+					<p>Mock Server Workflow Testing</p>
 					{data.workflow && <p> {data.workflow.edges[0].data.sourceLabel} TO {data.workflow.edges[0].data.targetLabel}</p>}
+
```
Comment (enZane) [inline services/dashboard/app/root.tsx:209]: Amazing! I just take a look and is awesome, just one thing, can you move this to another route? create one in the `app/routes` folder, and show up this component please

Comment (shurrini) [reply services/dashboard/app/root.tsx:209]: Sure thing! Just a question, should I move only the line you mentioned or the whole react flow?

Comment (enZane) [reply services/dashboard/app/root.tsx:209]: All the react flow! Github doesnt get multi-line hahaha

**Review by enZane — APPROVED** [review]

**Pull request merged**